security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -347,6 +355,10 @@ func (sc *SecretController) upsertSecret(saName, saNamespace string) {
 			k8sControllerLog.Infof("Secret %s/%s already exists, skip", saNamespace, GetSecretName(saName))
 			return
 		}
+		if errors.IsForbidden(err) {
```

Comment:
so this is a racing condition, that when citadel is about to update the secret, the namespace is actually being deleted?

mind adding a comment here explaining that above?